poet/sawtooth_poet/validator_registry/processor/handler.py

```diff
@@ -0,0 +1,333 @@
+        if anti_sybil_id == entry.key:
+            # The validator's old signup info is stale and needs to be revoked
+            validator_info = _get_validator_state(state, validator_id)
+
+            validator_info.registered = "revoked"
+            validator_info = validator_info.SerializeToString()
+            _set_validator_info(state, validator_id, validator_info)
+        return
+
+    # This is a new validator that needs to be added to the map
+    validator_map.entries.add(key=anti_sybil_id, value=validator_id)
+    address = _get_address('validator_map')
+    try:
+        addresses = list(state.set(
+            [StateEntry(address=address,
+                        data=validator_map.SerializeToString())],
+            timeout=STATE_TIMEOUT_SEC)
+        )
+
+    except FutureTimeoutError:
+        LOGGER.warning(
+            'Timeout occured on state.set([%s, <value>])', address)
+        raise InternalError(
+            'Unable to add {} to valdiator_list'.format(validator_id))
+
+    if len(addresses) != 1:
+        LOGGER.warning(
+            'Failed to save value on address %s', address)
+        raise InternalError(
+            'Unable to add {} to validator_map'.format(validator_id))
+
+    LOGGER.info("Validator id %s was added to the validator_map",
+                validator_id)
+
+
+def _set_validator_info(state, validator_id, validator_info):
+    address = _get_address(validator_id)
+    try:
+        addresses = list(state.set(
+            [StateEntry(address=address,
+                        data=validator_info)],
+            timeout=STATE_TIMEOUT_SEC)
+        )
+    except FutureTimeoutError:
+        LOGGER.warning(
+            'Timeout occured on state.set([%s, <value>])', address)
+        raise InternalError(
+            'Unable to set validator info for {}'.format(validator_id))
+
+    if len(addresses) != 1:
+        LOGGER.warning(
+            'Failed to save value on address %s', address)
+        raise InternalError(
+            'Unable to save validator info for {}'.format(validator_id))
+
+    LOGGER.info("Validator info was set for %s", validator_id)
+
+
+class ValidatorRegistryTransactionHandler(object):
+
+    __REPORT_PRIVATE_KEY_WIF = \
+        '5Jz5Kaiy3kCiHE537uXcQnJuiNJshf2bZZn43CrALMGoCd3zRuo'
+
+    def __init__(self):
+        self._report_private_key = \
+            signing.encode_privkey(
+                signing.decode_privkey(self.__REPORT_PRIVATE_KEY_WIF, 'wif'),
+                'hex')
+
+        self._report_public_key = signing.generate_pubkey(
+            self._report_private_key)
+
+    @property
+    def family_name(self):
+        return 'sawtooth_validator_registry'
+
+    @property
+    def family_versions(self):
+        return ['1.0']
+
+    @property
+    def encodings(self):
+        return ['application/protobuf']
+
+    @property
+    def namespaces(self):
+        return [VAL_REG_NAMESPACE]
+
+    def verify_signup_info(self,
```

Comment:
Currently we have no good way to have the transaction processor accesses the poet_enclave and since this code is untrusted python code, it can be done here.